File: Stratergy Pattern Implementation Code for Java/Player.java
package org.example;
public class Player {
    private AttackStrategy attackStrategy;

    // Method to set the attack strategy
    public void setAttackStrategy(AttackStrategy strategy) {
        this.attackStrategy = strategy;
    }

    // Method to execute the current attack strategy
    public void performAttack() {
        if (attackStrategy != null) {
            attackStrategy.attack();
        } else {
            System.out.println("No attack strategy set!");
        }
    }
}

File: Stratergy Pattern Implementation Code for Unity/Player.cs
using UnityEngine;

public class Player : MonoBehaviour
{
    private IAttackStrategy attackStrategy;

    // Method to set the attack strategy
    public void SetAttackStrategy(IAttackStrategy strategy)
    {
        attackStrategy = strategy;
    }

    // Method to execute the current attack strategy
    public void PerformAttack()
    {
        if (attackStrategy != null)
        {
            attackStrategy.Attack();
        }
        else
        {
            Debug.Log("No attack strategy set!");
        }
    }
}
